src/librustc_trans/back/link.rs

```diff
@@ -237,7 +237,16 @@ pub fn default_output_for_target(sess: &Session) -> config::CrateType {
 /// Checks if target supports crate_type as output
 pub fn invalid_output_for_target(sess: &Session,
                                  crate_type: config::CrateType) -> bool {
-    match (sess.target.target.options.dynamic_linking,
+    let requested_features = sess.opts.cg.target_feature.split(',');
```

Comment:
This addition duplicates part of the `add_configuration` function (librustc_driver/target_feature.rs) and should be refactored into its own function.